src/test/ui/issues/issue-23611-enum-swap-in-drop.rs

```diff
@@ -154,7 +154,7 @@ impl<'a> Drop for E<'a> {
         };
 
         if do_drop {
-            mem::replace(self, E::A(GaspA(f_a, 0xA3A0, log, D::new("drop", 6, log)), true));
+            let _ = mem::replace(self, E::A(GaspA(f_a, 0xA3A0, log, D::new("drop", 6, log)), true));
```

Comment:
FWIW, I think this could be changed to `*self = E::A(...)`, but seeing as this test is explicitly dealing with drop semantics, I didn't want to rock that boat.

Reply:
Can we instead `allow` the lint in this test? I am worried about a future change in the difference in drop semantics between an immediately dropping let binding and the free-standing expression.

Reply:
Ok, done.